server/static/js/term.js

```diff
@@ -166,6 +186,22 @@ function(RmcBackbone, _, $, _course, jqSlide, _user_course, _util) {
       this.template = _.template($('#profile-terms-tpl').html());
     },
 
+    addToShortlist: function(course) {
+      var term = this.termCollectionView.get('9999_99');
+      if (!term) {
+        // make shortlist if it does not exist
+        var shortlistModel = new TermModel({
+          'course_ids': [],
+          'program_year_id': 'None',
+          'id': '9999_99',
+          'name': 'Shortlist'
+        });
+        this.termCollectionView.addTerm(shortlistModel);
+        term = this.termCollectionView.get('9999_99');
+      }
+      term.addCourse(course, '9999_99');
+    },
+
```

Comment:
Its just to add a course to the shortlist and persist to the front-end. Previously, the button to add to shortlist and the shortlist are on different pages, so the data is automatically persisted to the front-end when reloading pages. However in this case, the recommendations and the shortlist are on the same page, so this is a way to show the change on the front-end.